src/sentry/llm/providers/openai.py

```diff
@@ -0,0 +1,49 @@
+from openai import OpenAI
+
+from sentry.llm.providers.base import LlmModelBase
+from sentry.llm.types import UseCaseConfig
+
+
+class OpenAIProvider(LlmModelBase):
+
+    provider_name = "openai"
+
+    def _complete_prompt(
+        self,
+        *,
+        usecase_config: UseCaseConfig,
+        prompt: str,
+        message: str,
+        temperature: float,
+        max_output_tokens: int,
+    ) -> str | None:
+        model = usecase_config["options"]["model"]
+        client = get_openai_client(self.provider_config["options"]["api_key"])
+
+        response = client.chat.completions.create(
+            model=model,
+            temperature=temperature
+            * 2,  # open AI temp range is [0.0 - 2.0], so we have to multiply by two
```

Comment:
Scaling the temperature feels a bit weird to me but I can get behind this use case for allowing it to be easy to swap out models–however temperature scales differently across different models, would it make sense to allow it to be configurable in the use case config too. This way when someone switches out a model they can adjust to the according temp for it